### PR TITLE
Learn page picture dialogs

### PR DESCRIPTION
**Pull request by bcbrennecke**
1 More PR before the code freeze.

1. Added dialogs for the learn page GIFs.  When you click on that screen area, the GIF pops out into a dialog.

2. Also fixed 404 error that was happening when loading Gifs from before.

- [x]  Functional

- [x] Functional

- [x] Code

- [x] Code

image

### CODE REVIEW

**Comment by Mctalian** [comment]
You sure your tests/files are being covered in the tests? I can't find the coverage results for learn.js

**Comment by YashdalfTheGray** [comment]
Your HTML in `pictureDialog.tpl.html` is formatted with 2 spaces instead of 4 spaces. And I also share the test concern with @Mctalian.

**Comment by bcbrennecke** [comment]
@YashdalfTheGray , I fixed the spaces.

**Comment by Mctalian** [comment]
karma.conf.js is the file you need to make sure the path to your file is
added there

**Comment by bcbrennecke** [comment]
Okay.  I will try to get to this tomorrow so we can get this closed.

**Comment by YashdalfTheGray** [comment]
Can you remove the GOT IT button and add an `X` button to the toolbar. It would make the picture take the focus of the dialog and remove the padding.

File: src/learn/learn.spec.js [hunk 1]
```diff
@@ -5,18 +5,21 @@
...
+describe('LearnCtrl', function() {
     "use strict";
 
     var ctrl, paginateSvc;
+    var scope;
 
     beforeEach(module('flintAndSteel'));
 
-    beforeEach(inject(function($controller, _paginateSvc_) {
+    beforeEach(inject(function($controller, _paginateSvc_, $rootScope) {
```
Comment (YashdalfTheGray) [inline src/learn/learn.spec.js]: remove `_paginateSvc_`.

File: src/learn/learn.js [hunk 2]
```diff
@@ -41,6 +41,22 @@ angular.module('flintAndSteel')
 
             $scope.LearnItems = [$scope.learnItem0, $scope.learnItem1, $scope.learnItem2, $scope.learnItem3, $scope.learnItem4];
 
+            $scope.showAdvanced = function(ev, item) {
+                $mdDialog.show({
+                    controller: 'PictureDialogController',
```
Comment (YashdalfTheGray) [inline src/learn/learn.js]: `PictureDialogController` should be `PictureDialogCtrl`.

File: src/learn/learn.js [hunk 3]
```diff
@@ -41,6 +41,22 @@ angular.module('flintAndSteel')
...
+                    templateUrl: 'learn/pictureDialog.tpl.html',
+                    parent: angular.element(document.body),
+                    targetEvent: ev,
+                    clickOutsideToClose: true,
+                    resolve: {
```
Comment (YashdalfTheGray) [inline src/learn/learn.js]: Change this to 
```javascript
resolve: {
   learnItem: item
}
```

File: src/learn/learn.spec.js [hunk 4]
```diff
@@ -4,23 +4,88 @@
...
+
+        it('should show a dialog window when called', function() {
+            scope.showAdvanced();
+            expect($mdDialog.show).toHaveBeenCalled();
+        });
+    });
+
+});
+
+describe('PictureDialogController', function() {
```
Comment (YashdalfTheGray) [inline src/learn/learn.spec.js]: Update the name here too.

File: src/learn/learn.spec.js [hunk 5]
```diff
@@ -4,23 +4,88 @@
...
+    "use strict";
+    var $controller, scope, $mdDialog, learnItem, ctrl;
+
+    beforeEach(module('flintAndSteel'));
+    beforeEach(inject(function($rootScope, _$controller_, _$mdDialog_) {
+        scope = $rootScope.$new();
+        $controller = _$controller_;
+        $mdDialog = _$mdDialog_;
+        learnItem = {title: "Test",
```
Comment (YashdalfTheGray) [inline src/learn/learn.spec.js]:
```javascript
learnItem = {
    title: "Test",
    imgSource: "assets/learn/learnToComment.gif",
    orderImg: "1",
    orderDesc: "2",
    description: "Test"
};
```

File: src/learn/pictureDialog.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* global angular */
+
+angular.module('flintAndSteel')
+.controller('PictureDialogController',
```
Comment (YashdalfTheGray) [inline src/learn/pictureDialog.js]: Update name here too.

File: src/learn/pictureDialog.tpl.html [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+                    <i class="material-icons icon-button-adjust">&#xE5CD;</i>
+                </md-button>
+            </div>
+        </md-toolbar>
+        <md-dialog-content style="learn-dialog-content" layout-align="center center" layout="row">
+            <img src={{item.imgSource}} class="learn-dialog-img">
+        </md-dialog-content>
+        <div class="md-actions" layout="row">
+            <span flex></span>
+            <md-button ng-click="answer()" class="learn-dialog-button" >
```
Comment (YashdalfTheGray) [inline src/learn/pictureDialog.tpl.html]: Why is there a still got it button here?

File: src/learn/pictureDialog.tpl.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        <md-toolbar>
+            <div  class="md-toolbar-tools">
+                <h4 class="learn-dialog-toolbar-padding">{{item.title}}</h4>
+                <span flex></span>
+                <md-button class="md-icon-button" ng-click="answer()" aria-label="Close" type="button">
+                    <i class="material-icons icon-button-adjust">&#xE5CD;</i>
+                </md-button>
+            </div>
+        </md-toolbar>
+        <md-dialog-content style="learn-dialog-content" layout-align="center center" layout="row">
```
Comment (YashdalfTheGray) [inline src/learn/pictureDialog.tpl.html]: This needs to be a `class` instead of a `style`.

**Pull request merged**